modules/basics/src/test/java/com/opengamma/strata/basics/date/HolidaySafeReferenceDataTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2022 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.date;
+
+import static com.opengamma.strata.collect.TestHelper.assertSerialization;
+import static com.opengamma.strata.collect.TestHelper.coverBeanEquals;
+import static com.opengamma.strata.collect.TestHelper.coverImmutableBean;
+import static java.time.DayOfWeek.SATURDAY;
+import static java.time.DayOfWeek.SUNDAY;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.DayOfWeek;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.basics.ImmutableReferenceData;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.ReferenceDataId;
+import com.opengamma.strata.basics.TestingReferenceDataId;
+
+/**
+ * Test {@link HolidaySafeReferenceData}.
+ */
+class HolidaySafeReferenceDataTest {
+
+  private static final ImmutableList<DayOfWeek> WEEKEND_DAYS = ImmutableList.of(SATURDAY, SUNDAY);
+  private static final HolidayCalendarId COMBINED_GBLO_EUTA_ID = HolidayCalendarId.of("GBLO+EUTA");
+  private static final HolidayCalendarId LINKED_GBLO_EUTA_ID = HolidayCalendarId.of("GBLO~EUTA");
+  private static final ImmutableHolidayCalendar GBLO_EUTA_CAL = ImmutableHolidayCalendar.combined(
+      createCalendar(HolidayCalendarIds.GBLO), createCalendar(HolidayCalendarIds.EUTA));
+  private static final TestingReferenceDataId ID1 = new TestingReferenceDataId("1");
+  private static final TestingReferenceDataId ID2 = new TestingReferenceDataId("2");
+  private static final Number VAL1 = 1;
+  private static final Number VAL2 = 2;
+
+  private static ImmutableHolidayCalendar createCalendar(HolidayCalendarId id) {
+    return ImmutableHolidayCalendar.of(id, ImmutableList.of(), WEEKEND_DAYS);
+  }
+
+  @Test
```

Comment:
I would like to see tests using `getValue` or `findValue` in the super-interface as that is the main API that needs to be satisfied. eg. searching for GBLO+EUTA where GBLO is in reference data but EUTA is not should default EUTA and then combine it.